### PR TITLE
drivers: dma stm32 keep the channel busy if overriden

### PR DESCRIPTION
**Pull request by FRASTM**
Actually the busy flag of a dma stream is no pertinent when this stream is overridden by the HAL.

After merging the https://github.com/zephyrproject-rtos/zephyr/pull/44133, 
we found that  running the tests/drivers/flash on the disco board stm32l475, fails.
This test is running on the qspi  flash with DMA transfer of the stm32l475 disco kit. This flash_stm32_qspi driver is using the HAL to control the DMA transfer with HAL_DMA_Init and HAL_DMA_IRQHandler. That's the purpose of the `hal_override` flag of the stream (channel). Both stm32l4xx_hal_dma functions do not manage the DMA  irq as the stm32l4xx_ll_dma does. This makes DMA irq triggered even when the channel is no busy anymore.

This fix is ignoring the status of a stream busy  if this one was overridden (controlled) by the HAL.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: drivers/dma/dma_stm32.c [hunk 1]
```diff
@@ -563,9 +572,12 @@ DMA_STM32_EXPORT_API int dma_stm32_start(const struct device *dev, uint32_t id)
 	}
 
 	dma_stm32_clear_stream_irq(dev, id);
-
 	stm32_dma_enable_stream(dma, id);
 
+	/* When starting the dma, the stream is busy */
+	stream = &config->streams[id];
```
Comment (teburd) [inline drivers/dma/dma_stm32.c:578]: Is it possible an irq is fired (since the stream was enabled above) before stream->busy is set?

Comment (FRASTM) [reply drivers/dma/dma_stm32.c:578]: Yes, that could, so inverting setting busy then enabling (when starting and reloading)

**Comment by FRASTM** [comment]
setting busy before enabling the DMA channel enabling (when starting and reloading)

**Review by teburd — APPROVED** [review]
Looks good to me

**Review by erwango — APPROVED** [review]

**Pull request merged**